src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClients/ConfigurationClientState.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Constants;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions;
+using System;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.ConfigurationClients
+{
+    internal class ConfigurationClientState
```

Comment:
I had a comment about tuple in `FailoverSupportedClient`. I'd actually think this might be a good candidate as a private type in FailoverSupportedClient to replace the tuple.